charts/external-dns/templates/deployment.yaml

```diff
@@ -132,9 +137,51 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
-      {{- if or .Values.secretConfiguration.enabled .Values.extraVolumes }}
+        {{- if contains "/" $providername }}
+        - name: provider
+          {{- with .Values.securityContext }}
+          securityContext:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          image: {{ tpl $providername $ }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- with .Values.provider.env }}
+          env:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          args:
+          {{- range .Values.provider.args }}
+            - {{ tpl . $ }}
+          {{- end }}
+          ports:
+            - name: provider-metrics
+              protocol: TCP
+              containerPort: 8080
```

Comment:
Is this port expected on all webhook providers ?
FTM, I have seen they are using 8888 by default, to serve provider method, not metrics.

Reply:
That is my proposal. We'll want them to have a separate port for metrics because security depends on the provider methods not being accessible outside of the pod. I don't see any benefit in letting webhook providers choose different metrics ports.